pkg/controller/operators/olm/operator.go

```diff
@@ -1084,39 +1082,27 @@ func (a *Operator) transitionCSVState(in v1alpha1.ClusterServiceVersion) (out *v
 			return
 		}
 
-		// If we are a leaf, we should requeue the replacement for processing
+		// If there is a succeeded replacement, mark this for deletion
```

Comment:
It simplifies everything if we step 1 by 1 instead of marking multiple for deletion at once. This addressed a flakey test and let us delete a chunk of code.